client/post-editor/post-editor.jsx

```diff
@@ -820,6 +816,11 @@ export const PostEditor = React.createClass( {
 
 	onSaveSuccess: function( message, action, link ) {
 		const post = PostEditStore.get();
+		const isNotPrivateOrIsConfirmed = ( 'private' !== post.status ) || ( 'closed' !== this.state.confirmationSidebar );
```

Comment:
It looks like `onPublish()` is setting the `confirmationSidebar` state to `publish` before this check fires - thus always true.